provision/acc_provision/acc_provision.py

```diff
@@ -115,8 +115,23 @@
             },
         },
     },
+    "cloudfoundry-1.0": {
+        "desc": "CloudFoundry cf-deployment 1.x",
+        "default_version": "latest",
+        "config": {
```

Comment:
Shouldn't you need to set something in here to enable CF mode?

The intention here is that any differences between versions are expressed in terms of config options, and the flavors just initialize the default values for all the config options.  For example if kubeapi_vlan isn't used for CF, we have an option "use_kubeapi_vlan" and set to true for Kube and false for CF.  Then the code that requires divergent behavior is expressed in the form of one of these options.

This means that if future versions of CF and kube differ in other ways we can always express this appropriately using the default config in the flavor.

I guess this was not as clear as I thought it might be from the code; maybe there needs to be a comment :-)

Reply:
That makes things clearer. Instead of adding config options for the different behavior I desire, I'm going to add "flavor options" which will be different from "config". These will not be user configurable, unlike the config options. The intention of these flavor options is to alter the execution flow for different flavors. Example of such options are use_kubeapi_vlan, template_generator_function etc

Reply:
There's already a lot of things in there that are not really things the user is likely to configure except by setting the flavor, such as most of the OpenShift options.  So not really needed to separate it I think.

Reply:
I think a distinction needs to be made between not likely to configure versus must not be able to modify. I think some of the confusion with the flavor stuff stems from overloading the purpose of config.

Reply:
well most of the config options are already such that they need to be set correctly or the system won't work.  I don't see why cloudfoundry config items should use a separate method.

The key thing to note is that I'd expect, e.g. a future version of OpenShift to require fewer workarounds so an advanced user might actually set config options to override the default flavor to run on a newer version.

Reply:
If you're proposing a broader refactor that's a different question of course; I'm just saying I'd like to keep them consistent.